webapi/src/main/scala/org/knora/webapi/util/DateUtilV1.scala

```diff
@@ -102,33 +102,69 @@ object DateUtilV1 {
         // When setting the date, set time to noon (12) as JDC would contain a fraction otherwise:
         // "Julian Days can also be used to tell time; the time of day is expressed as a fraction of a full day, with 12:00 noon (not midnight) as the zero point."
         // From: https://docs.kde.org/trunk5/en/kdeedu/kstars/ai-julianday.html
+
+        // set leniency setting to false as leniency would allow for invalid dates such as Feb 30.
+        // https://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html#setLenient-boolean-
+
+        // call get method on calendar after setting the date, an exception is thrown if the date is invalid
+        // https://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html#get-int-
+
         dateSegments.length match {
             case 1 => // year precision
-                val intervalStart = new GregorianCalendar
-                intervalStart.setGregorianChange(changeDate)
-                intervalStart.set(dateSegments(0).toInt, 0, 1, 12, 0, 0) // January 1st of the given year
 
-                val intervalEnd = new GregorianCalendar
-                intervalEnd.setGregorianChange(changeDate)
-                intervalEnd.set(dateSegments(0).toInt, 11, 31, 12, 0, 0) // December 31st of the given year
+                try {
+                    val intervalStart = new GregorianCalendar
+                    intervalStart.setLenient(false)
+                    intervalStart.setGregorianChange(changeDate)
+                    intervalStart.set(dateSegments(0).toInt, 0, 1, 12, 0, 0) // January 1st of the given year. Attention: in java.util.Calendar, month count starts with 0
+                    intervalStart.get(0)
```

Comment:
Please add a comment here to explain that you're calling `get(0)` because this causes all calendar fields to be validated, and throws an exception if any fields have out-of-range values.